File: Media-Pembelajaran-Administrasi-Infrastruktur-Jaringan-master/app/src/main/java/com/fano/administrasiinfrastrukturjaringan/ListVideoActivity.java
package com.fano.administrasiinfrastrukturjaringan;

import androidx.appcompat.app.AppCompatActivity;
import androidx.cardview.widget.CardView;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.ImageButton;
import android.widget.TextView;

public class ListVideoActivity extends AppCompatActivity {
    ImageButton ibBack;
    TextView tvToolbar;
    CardView cvAij;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_list_video);

        tvToolbar = findViewById(R.id.tv_toolbar);
        ibBack = findViewById(R.id.ib_back);
        cvAij = findViewById(R.id.cv_aij);

        tvToolbar.setText("Daftar Video");

        ibBack.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                onBackPressed();
            }
        });

        cvAij.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(getApplicationContext(), VideoActivity.class);
                intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
                getApplicationContext().startActivity(intent);
            }
        });

    }
}

File: AdministrasiSistemJaringan/app/src/main/java/id/topan/administrasisistemjaringan/ListVideoActivity.kt
package id.topan.administrasisistemjaringan

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.androidnetworking.AndroidNetworking
import com.androidnetworking.common.Priority
import com.androidnetworking.error.ANError
import com.androidnetworking.interfaces.JSONObjectRequestListener
import id.topan.administrasisistemjaringan.adapter.VideoAdapter
import id.topan.administrasisistemjaringan.model.ModelVideo
import id.topan.administrasisistemjaringan.network.ApiServer
import org.json.JSONObject

class ListVideoActivity : AppCompatActivity() {

    private var video : ArrayList<ModelVideo> = arrayListOf()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_list_video)
        AndroidNetworking.initialize(this)

        val rvVideo : RecyclerView = findViewById(R.id.rv_video)
        rvVideo.layoutManager = LinearLayoutManager(this)
        rvVideo.setHasFixedSize(true)

        getVideo(rvVideo)


    }

    private fun getVideo(rvVideo: RecyclerView) {
        AndroidNetworking.get(ApiServer.site_url+"getDataVideo")
            .setPriority(Priority.MEDIUM)
            .build()
            .getAsJSONObject(object : JSONObjectRequestListener{
                override fun onResponse(response: JSONObject?) {
                    if (response != null) {
                        Log.d("sukses", "code :"+response)
                        if(response.getString("status") == "200"){
                            val data = response.getJSONArray("data")
                            video.clear()
                            for (i in 0 until data.length()){
                                val item = data.getJSONObject(i)
                                video.add(
                                    ModelVideo(
                                    item.getString("id_video"),
                                    item.getString("judul_video"),
                                    item.getString("link")
                                ))
                                val adapterVideo = VideoAdapter(video)
                                rvVideo.adapter = adapterVideo
                                adapterVideo.notifyDataSetChanged()
                            }
                        }
                    }
                }

                override fun onError(anError: ANError?) {
                    Log.d("eror", "eror : "+anError)
                }
            })
    }
}